server/routes/pokemonRoutes.js

```diff
@@ -159,5 +166,61 @@ module.exports = {
 				});
 			}
 		});
+
+		app.post(props.routes.root + props.routes.rename, (req, res, next) => {
+			const endpoint = props.routes.root + props.routes.rename;
+			log.debug(
+				{username: req.body.username},
+				'POST request to ' + endpoint);
+
+			if(!req.body.hasOwnProperty('username')){
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.username}, req.body.username, endpoint);
+			} else if (!req.body.hasOwnProperty('password')) {
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.password}, req.body.username, endpoint);
+			} else if (!req.body.hasOwnProperty('type')) {
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.type}, req.body.username, endpoint);
+			} else if (!req.body.hasOwnProperty('pokemon_id')) {
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.pokemon_id}, req.body.username, endpoint);
+			} else if (!req.body.pokemon_id.hasOwnProperty('high')){
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.invalid_pokemon_id}, req.body.username, endpoint);
+			} else if (!req.body.pokemon_id.hasOwnProperty('low')){
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.invalid_pokemon_id}, req.body.username, endpoint);
+			} else if (!req.body.pokemon_id.hasOwnProperty('unsigned')){
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.invalid_pokemon_id}, req.body.username, endpoint);
+			} else if (!req.body.hasOwnProperty('nickname')) {
+				expressUtils.sendResponse(res, next, 400, {error: props.errors.nickname}, req.body.username, endpoint);
+			}
+			else {
+				req.body.pokemon_id.high = Number(req.body.pokemon_id.high);
+				req.body.pokemon_id.low = Number(req.body.pokemon_id.low);
+				req.body.pokemon_id.unsigned = req.body.pokemon_id === "true" ? true : false;
+
+				const client = new pogobuf.Client();
+
+				let login = null;
+
+				if(req.body.type.toLowerCase() === 'google'){
+					login = new pogobuf.GoogleLogin();
+				} else {
+					login = new pogobuf.PTCLogin();
+				}
+
+				login.login(req.body.username, req.body.password).then(token => {
+					client.setAuthInfo(req.body.type.toLowerCase(), token);
+
+					client.init().then(() => {
+						client.nicknamePokemon(req.body.pokemon_id, req.body.nickname).then(releaseResponse => {
```

Comment:
"releaseResponse" should be changed to "nicknameResponse" or something that makes sense given the operation